streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -590,7 +590,14 @@ boolean runLoop() {
                 log.warn("Detected the states of tasks " + e.corruptedTasks() + " are corrupted. " +
                          "Will close the task as dirty and re-create and bootstrap from scratch.", e);
                 try {
+                    // check if any active task got corrupted. We will trigger a rebalance in that case.
+                    // once the task corruptions have been handled
+                    Set<Task> corruptedActiveTasks = taskManager.anyActiveTasksCorrupted(e.corruptedTasks());
                     taskManager.handleCorruption(e.corruptedTasks());
+                    if (!corruptedActiveTasks.isEmpty()) {
```

Comment:
Note that we actually only clear the corrupted state in EOS applications (don't ask me whether I agree with this 😒 ). So we should add a `&& eosEnabled` to the condition for rebalancing.

Reply:
done